tests/TestCase/TestSuite/IntegrationTestCaseTest.php

```diff
@@ -143,6 +144,28 @@ public function testRequestBuildingQueryParameters()
         $this->assertEquals('yes', $request->query('archived'));
     }
 
+    public function testCookieWithNoEncription()
+    {
+        $this->cookie('KeyOfCookie', 'CookieComponent is not used');
+        $request = $this->_buildRequest('/tasks/view', 'GET', []);
+        $this->assertEquals('CookieComponent is not used', $request->cookies['KeyOfCookie']);
+    }
+
+    public function testCookieWithEncriptionFalse()
+    {
+        $this->cookie('KeyOfCookie', ['Array is', 'json-encoded', 'by CookieComponent'], false);
+        $request = $this->_buildRequest('/tasks/view', 'GET', []);
+        $this->assertEquals('["Array is","json-encoded","by CookieComponent"]', $request->cookies['KeyOfCookie']);
+    }
+
+    public function testCookieWithEncriptionAes()
+    {
+        Security::salt('foo!foo!foo!foo!foo!foo!foo!foo!');
+        $this->cookie('KeyOfCookie', 'Encrypted by CookieComponent', 'aes');
+        $request = $this->_buildRequest('/tasks/view', 'GET', []);
+        $this->assertStringStartsWith('Q2FrZQ==.', $request->cookies['KeyOfCookie']);
```

Comment:
It would be good to make sure a controller action can decode the cookie as well.